### ISSUE TITLE
Can NOT stop timer when playing in a BG thread

### ISSUE THREAD

**Issue by kthtes** [wang-bin/QtAV#723]
**Platform:**
Qt 5.7.0 + macOS (this happens in Ubuntu/Win too)
QtAV 1.11.0

**Steps to reproduce this bug:**

1. Create a new QML project TestQtav.pro by choosing **Qt Quick Application**

2. In TestQtav.pro:
`QT += qml quick av`

3. In main.qml:
```
import QtQuick 2.7
import QtQuick.Controls 2.0
import QtQuick.Window 2.2
import QtAV 1.6

Window {
    visible: true
    width: 640
    height: 480
    title: qsTr("Hello World")

    MainForm {
        anchors.fill: parent
        mouseArea.onClicked: {
            Qt.quit();
        }
    }

    Button{
        text: 'Test QtAV Thread'
        onClicked: {
            eval('playInBgThread()')
        }
    }

    AVPlayer{
        id: myPlayer
        autoPlay: true
    }

    function playInBgThread(){
        myPlayer.source='480.m4a'  // '480.m4a' is a short voice file of 2~3 seconds
    }
}
```

**Note:**
1. The code above let AVPlayer run in an independent **eval** thread other than main thread.
2. When the short 480.m4a ends, in console, it shows:
> QObject::killTimer: Timers cannot be stopped from another thread

I think it may be some thread-related bug(s).

**Comment by kthtes** [comment]
Thanks for fixing it.

I'll test after #724 is fixed and close both of them by then.

**Comment by kthtes** [comment]
I've tested this bug-fix and #724. It's good now. I'm closing the 2 issues.

Thanks!
